vault/testing.go

```diff
@@ -390,6 +392,10 @@ func testCoreAddSecretMount(t testing.T, core *Core, token string) {
 
 }
 
+func TestCoreUnsealedBackendBenchmarking(t gotesting.TB, backend physical.Backend) (*Core, [][]byte, string) {
```

Comment:
I'm not seeing why these are needed?

Reply:
Ah, so github.com/mitchellh/go-testing-interface v1.14.1 has updated the T interface to include a Parallel() method. This method does not exist in the TB interface in the native testing library, and so methods that were originally using github.com/mitchellh/go-testing-interface v1.14.1's T interface and being called by functions that needed to use the TB interface because of benchmarking tests no longer worked. 

The solution I thought of was just to wrap the methods that require the github.com/mitchellh/go-testing-interface v1.14.1 T interface with a "Benchmarking" function that takes in a TB interface parameter and casts it to a github.com/mitchellh/go-testing-interface v1.14.1 T interface. In the case where a native Golang testing T struct is used, everything will work fine. In the case where a B struct is used, B will get cast to TB, which can be cast to T because T has all the methods in TB, but not vice versa.

Reply:
Ok, thanks for explaining.  I'm not thrilled with this solution, but it'll do and I don't have a better one off the top of my head.  I mean we could PR a TB type into go-testing-interface and use that in all the test helper signatures, but I don't want to block this PR just for that.